src/ast.c

```diff
@@ -208,11 +208,54 @@ static value_t fl_nothrow_julia_global(fl_context_t *fl_ctx, value_t *args, uint
     return b != NULL && jl_atomic_load_relaxed(&b->value) != NULL ? fl_ctx->T : fl_ctx->F;
 }
 
+arraylist_t parsed_method_stack; // for keeping track of which methods are being parsed
+uv_mutex_t counter_table_lock;
+htable_t counter_table; // map from module_name -> inner htable; inner htable maps from char * -> uint32_t
```

Comment:
This could go into `jl_module_t`, which already has a lock (and an atomic field called `counter`; not sure if that's related?). I think that will reduce a lookup here and let us avoid having yet another couple of globals.

Reply:
Have made it per-module.